src/util/logging.go

```diff
@@ -32,14 +42,86 @@ func setLoggingOnLogger(name string, logger *log.Logger) {
 			return "", fmt.Sprintf("%s:%d", formatFilePath(f.File), f.Line)
 		},
 	})
+}
 
+func setNullWriter(logger *logrus.Logger) {
+	logger.SetOutput(ioutil.Discard)
 }
 
 func formatFilePath(path string) string {
 	arr := strings.Split(path, "/")
 	return arr[len(arr)-1]
 }
 
-func SetLogging(name string) {
-	setLoggingOnLogger(name, log.StandardLogger())
+func setJournalLogging(logger *logrus.Logger, journalWriter IJournalWriter) {
+	logger.AddHook(newJournalHook(journalWriter))
+}
+
+//go:generate mockery -name IJournalWriter -inpkg
+type IJournalWriter interface {
+	Print(p journald.Priority, format string, a ...interface{}) error
+}
+
+type journalHook struct{
+	writer IJournalWriter
+}
+
+type JournalWriter struct {}
+
+func (*JournalWriter) Print(p journald.Priority, format string, a ...interface{}) error {
+	return journald.Print(p, format, a...)
+}
+
+func newJournalHook(writer IJournalWriter) *journalHook {
+	return &journalHook{writer:writer}
+}
+
+func (hook *journalHook) getPriority(entry *logrus.Entry) journald.Priority {
+	switch entry.Level {
+	case logrus.TraceLevel, logrus.DebugLevel:
+		return journald.PriorityDebug
+	case logrus.InfoLevel:
+		return journald.PriorityInfo
+	case logrus.WarnLevel:
+		return journald.PriorityWarning
+	case logrus.ErrorLevel:
+		return journald.PriorityErr
+	case logrus.FatalLevel:
+		return journald.PriorityCrit
+	case logrus.PanicLevel:
+		return journald.PriorityEmerg
+	default:
+		return journald.PriorityInfo
+	}
+}
+
+func (hook *journalHook) Fire(entry *logrus.Entry) error {
+	line, err := entry.String()
+	if err != nil {
+		return err
+	}
+	return hook.writer.Print(hook.getPriority(entry), line)
+}
+
+func (hook *journalHook) Levels() []logrus.Level {
+	return logrus.AllLevels
+}
+
+func setLogging(logger *logrus.Logger, journalWriter IJournalWriter, name string, textLogging, journalLogging bool) {
```

Comment:
@ori-amizur why not log driver?

https://docs.docker.com/config/containers/logging/configure/

Isnt it over complication?

Reply:
If we are using a container, then stdout, and stderr are used to send the output to the parent process.

Reply:
Therefore, docker log driver cannot be used because it relies on stdout/stderr of the process,